automation_tools/satellite6/upgrade/tasks.py

```diff
@@ -401,9 +408,176 @@ def sync_capsule_tools_repos_to_upgrade(admin_password=None):
+        print('Clients count to generate on Docker cannot be Zero !!')
+        sys.exit(1)
+    rhevm_client = get_rhevm_client()
+    instance_name = 'upgrade_dockered_clients_image_DND'
+    template_name = 'upgrade_dockered_image'
+    satellite_hostname = os.environ.get('RHEV_SAT_HOST')
+    ak = os.environ.get('RHEV_CLIENT_AK')
+    result = {}
+    docker_vm_status = rhevm_client.vms.get(
+        name=instance_name).get_status().get_state()
+    if docker_vm_status != 'up':
+        print('Docker VM is not up. Turning on, please wait ....')
+        create_rhevm_instance(instance_name, template_name)
+        wait_till_rhev_instance_status(instance_name, 'up')
+        print('Docker VM is now up !! Generating clients ....')
+    for count in range(int(clients_count)):
+        hostname = '{0}DockerClient{1}'.format(count, client_os)
+        container_id = run(
+            'docker run -d -h {0} -v /dev/log:/dev/log -e "SATHOST={1}" '
+            '-e "AK={2}" upgrade:{3}'.format(
+                hostname, satellite_hostname, ak, client_os))
+        result[hostname] = container_id
+    rhevm_client.disconnect()
+    return result
+
+
+def refresh_subscriptions_on_docker_clients(container_ids):
+    """Refreshes subscription on docker containers which are satellite clients
+
+    :param list container_ids: The list of container ids onto which
+    subscriptions will be refreshed
+    """
+    if container_ids is list:
+        for container_id in container_ids:
+            run('docker exec {} subscription-manager refresh'.format(
+                container_id))
+            run('docker exec {} yum clean all'.format(container_id))
+    else:
+        run('docker exec {} subscription-manager refresh'.format(
+            container_ids))
+        run('docker exec {} yum clean all'.format(container_ids))
+
+
+def sync_tools_repos_to_upgrade(client_os, hosts):
+    """This syncs tools repo in Satellite server and also attaches
+    the new tools repo subscription onto each client
+
+    :param string client_os: The client OS of which tools repo to be synced
+        e.g: rhel6, rhel7
+    :param list hosts: The list of capsule hostnames to which new capsule
+        repo subscription will be attached
+
+    Following environment variable affects this function:
+
+    TOOLS_URL_{client_os}
+        The url of tools repo from latest satellite compose.
+    FROM_VERSION
+        Current Satellite version - to differentiate default organization.
+        e.g. '6.1', '6.0'
+
+    Personal Upgrade Env Vars:
+
+    CLIENT_SUBSCRIPTION
+        List of cv_name, environment, ak_name attached to subscription of
+        client in defined sequence
+
+    Rhevm upgrade Env Vars:
+
+    RHEV_CLIENT_CV
+        The CV name used in client subscription
+    RHEV_CLIENT_ENVIRONMENT
+        The environment name used in client subscription
+    RHEV_CLIENT_AK
+        The AK name used in client subscription
+    """
+    client_os = client_os.upper()
```

Comment:
The os installed in client e.g rhel7, rhel6 of satellite.
To get the Tools repo env var correct i uppered it.
```TOOLS_URL_{client_os}```